realm/realm-library/src/androidTest/java/io/realm/MutableRealmIntegerTests.java

```diff
@@ -366,12 +363,12 @@ private void testNullability(MutableRealmIntegerTypes c1) {
             c1.columnMutableRealmInteger.increment(5);
             fail("Attempt to increment a null valued MutableRealmInteger should throw NPE");
         }
-        catch (NullPointerException ignore) { }
+        catch (IllegalArgumentException ignore) { }
```

Comment:
Isn't `IlegalStateException` more accurate? Also the message for `fail()` needs to be changed.